docs/deployment/docker.md

```diff
@@ -200,12 +201,13 @@ RUN mkdir packages/backend/dist/skeleton packages/backend/dist/bundle \
 # Stage 3 - Build the actual backend image and install production dependencies
 FROM node:16-bullseye-slim
 
-# Install sqlite3 dependencies. You can skip this if you don't use sqlite3 in the image,
+# Install sqlite3 and isolated-vm dependencies.
+# You can remove libsqlite3-dev if you don't use sqlite3 in the image,
 # in which case you should also move better-sqlite3 to "devDependencies" in package.json.
 RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
     --mount=type=cache,target=/var/lib/apt,sharing=locked \
     apt-get update && \
-    apt-get install -y --no-install-recommends libsqlite3-dev python3 build-essential && \
+    apt-get install -y --no-install-recommends libsqlite3-dev python3 g++ build-essential && \
```

Comment:
As isolated-vm fully operational is required for Scaffolder Backend Plugin to run, as this Plugin belongs to the basic create-app, _python3_ _g++_ and _build-essential_ are required if the Backstage instance is to run along side sqlite3 or with an external Postgres.
_python3_ _g++_ and _build-essential_ are to be installed separately, to prevent devs setuping their Dockerfile without Postgres sadly having [this error](https://github.com/backstage/backstage/issues/18370).

Reply:
They don't need to be installed separately, this works as I tested both using Postgres and sqlite3. Postgres uses the [node-postgres](https://github.com/brianc/node-postgres) package `pg` which is pure JS and doesn't ned any external deps.

Reply:
Rereading this I don't actually understand what you are asking, sorry 😢

Reply:
With all due respect, the whole point of this [issue](https://github.com/backstage/backstage/issues/18370) was caused because a "RUN apt-get install -y --no-install-recommends python3 g++ build-essential && yarn config set python /usr/bin/python3" is still required without sqlite3, for isolated-vm to be able to work properly (and not having a docker build failure).

Reply:
> Rereading this I don't actually understand what you are asking, sorry 😢

maybe i commit a Dockerfile myself, with my point and all the remaining Boy Scout Rule cleaning ?

Reply:
My apologies, I just don't understand what you are saying. Are you saying to adjusts the comment? Or to break apart. the `RUN`?

Reply:
@Deffohn, I'm terribly sorry, I took my time off on a Saturday to test this documentation and make the needed updates. I'm not sure why you seem to be upset with me about this. I don't understand what you mean by "Boy Scout Rule cleaning"?

Reply:
I advise splitting this RUN apart because Python, g++ and build-essential are still required whether you setup sqlite3 or not

Reply:
Ok, now I understand what you are saying. But would just removing `libsqlite3-dev` from the `RUN` not accomplish the same thing?

Reply:
> @Deffohn, I'm terribly sorry, I took my time off on a Saturday to test this documentation and make the needed updates. I'm not sure why you seem to be upset with me about this. I don't understand what you mean by "Boy Scout Rule cleaning"?

The multi-stage Dockerfile can be cleaned and optimized (dividing build time by 2 I think), besides allowing all new Backstage dev who will setup its own instance without sqlite3 running into this trap of not having proper requirements for isolated-vm.

> Ok, now I understand what you are saying. But would just removing `libsqlite3-dev` from the `RUN` not accomplish the same thing?

For now, your current correction is not enough : by line 205 "# You can remove libsqlite3-dev if you don't use sqlite3 in the image,", a dev setting up its own Backstage deployment would crash and have to search in Github issues or in old Release notes to solve this installation error.

=> Maybe let me take time by Sunday morning (Europe time for me) to improve this all ?

Reply:
Alright, I broken up the `RUN` statements as per your suggestions. This does mean you have another layer but I think in the long run that's probably not a huge issue. For those that it is they will just removed what they don't need

Reply:
@Deffohn should I abandon this PR then? That's fine with me. 👍

Reply:
> @Deffohn should I abandon this PR then? That's fine with me. 👍

It be better to rewrite completly those 3 Dockerfile deployments, maybe just keep this PR with your modifications, as they are informative enough for devs reading the Backstage doc.
Just don't forget doing the same by line 60 for Host Build and line 15 of backend only Dockerfile

Reply:
Sure, at this point I'm just going to leave it as is then, everything is working the way it's written right now which for me is the most important part. If you have more suggestions tomorrow I'm happy to make them. The Backstage maintainers will probably give it a look on Monday

Reply:
> Sure, at this point I'm just going to leave it as is then, everything is working the way it's written right now which for me is the most important part. If you have more suggestions tomorrow I'm happy to make them. The Backstage maintainers will probably give it a look on Monday

Maybe i'll try my own first PR on Backstage by the afternoon optimizing it.